test/SemVer.unittest.js

```diff
@@ -0,0 +1,513 @@
+"use strict";
+
+const {
+	parseVersion,
+	versionToString,
+	versionLt,
+	parseRange,
+	rangeToString,
+	satisfy
+} = require("../lib/util/semver");
+
+describe("SemVer", () => {
+	it("should parseVersion correctly", () => {
+		expect(parseVersion("1")).toEqual([1]);
+		expect(parseVersion("1.2.3")).toEqual([1, 2, 3]);
+		expect(parseVersion("1.2.3.4.999")).toEqual([1, 2, 3, 4, 999]);
+		// eslint-disable-next-line no-sparse-arrays
+		expect(parseVersion("1.2.3-beta")).toEqual([1, 2, 3, , "beta"]);
+		// eslint-disable-next-line no-sparse-arrays
+		expect(parseVersion("1.2.3-beta.1.2")).toEqual([1, 2, 3, , "beta", 1, 2]);
+		// eslint-disable-next-line no-sparse-arrays
+		expect(parseVersion("1.2.3-alpha.beta-42")).toEqual([
+			1,
+			2,
+			3,
+			,
+			"alpha",
+			"beta-42"
+		]);
+		// eslint-disable-next-line no-sparse-arrays
+		expect(parseVersion("1.2.3-beta.1.alpha.0+5343")).toEqual([
+			1,
+			2,
+			3,
+			,
+			"beta",
+			1,
+			"alpha",
+			0,
+			[],
+			5343
+		]);
+		expect(parseVersion("1.2.3+5343.beta+1")).toEqual([
+			1,
+			2,
+			3,
+			[],
+			5343,
+			"beta+1"
+		]);
+		expect(parseVersion("1.2.3+5343.beta+1")).toEqual([
+			1,
+			2,
+			3,
+			[],
+			5343,
+			"beta+1"
+		]);
+	});
+
+	describe("versionToString", () => {
+		const cases = [
+			"0.0.0",
+			"0.0.1",
+			"0.1.2",
+			"1.2.3",
+			"1.0.0",
+			"1.2.3.4.5.6.7.8",
+			"1.2",
+			"1",
+			"1.2.3-beta",
+			"1.2.3-beta.15",
+			"1.2.3-beta.15+45",
+			"1.2.3-beta.15+45.67",
+			"1.2.3+45",
+			"1.2.3+45.67",
+			"1.2.3-beta-gamma-test+x+y+z"
+		];
+		for (const key of cases) {
+			it(`should ${key} stringify correctly`, () => {
+				expect(versionToString(parseVersion(key))).toEqual(key);
+			});
+		}
+	});
+
+	describe("versionLt", () => {
+		const cases = [
+			"1 < 2",
+			"99 < 100",
+			"1 < 1.2",
+			"1 < 1.2.3",
+			"1.2 < 1.2.3",
+			"1.2.2 < 1.2.3",
+			"1.1.3 < 1.2.0",
+			"1.1.3 < 2.0.0",
+			"1.1.3 < 2",
+			"1.1.3 < 2.0",
+			"1.2.3 < 1.2.3+0",
+			"1.2.3+23 < 1.2.3+123",
+			"1.2+2 < 1.2.3+1",
+			"1.2.3-beta < 1.2.3",
+			"1.2.3 < 1.2.4-beta",
+			"1.2.3 < 1.3.0-beta",
+			"1.2.3 < 2.0.0-beta",
+			"1.2.3-alpha < 1.2.3-beta",
+			"1.2.3-beta < 1.2.3.1",
+			"1.2.3-beta < 1.2.3-beta.0",
+			"1.2.3-beta.0 < 1.2.3-beta.1",
+			"1.2.3-0 < 1.2.3-beta",
+			"1.2.3-beta < 1.2.3-beta+123",
+			"1.2.3-beta+123 < 1.2.3-beta+234",
+			"1.2.3-beta+99 < 1.2.3-beta+111"
```

Comment:
For completeness, might as well add the official semver example as well. I also added one more tests to test prerelease vs build:

```suggestion
			"1.2.3-beta+99 < 1.2.3-beta+111",
			"1.2.3-beta < 1.2.3+1",
			"1.0.0-alpha < 1.0.0-alpha.1",
			"1.0.0-alpha.1 < 1.0.0-alpha.beta",
			"1.0.0-alpha.beta < 1.0.0-beta",
			"1.0.0-beta < 1.0.0-beta.2",
			"1.0.0-beta.2 < 1.0.0-beta.11",
			"1.0.0-beta.11 < 1.0.0-rc.1",
			"1.0.0-rc.1 < 1.0.0"
```